sqlite.go

```diff
@@ -178,12 +178,15 @@ func openConn(path string, flags OpenFlags) (*Conn, error) {
 	return conn, nil
 }
 
-// Close closes the database connection using sqlite3_close_v2.
-// https://www.sqlite.org/c3ref/close.html
+// Close closes the database connection using sqlite3_close and finalizes
+// persistent prepared statements. https://www.sqlite.org/c3ref/close.html
 func (conn *Conn) Close() error {
 	conn.cancelInterrupt()
 	conn.closed = true
-	res := C.sqlite3_close_v2(conn.conn)
+	for _, stmt := range conn.stmts {
+		stmt.Finalize()
```

Comment:
What do you think about returning an error if a statement is in the middle of returning results? We could check `stmt.lastHasRow` here.

For a later commit.

Reply:
I think it's not unexpected for an application to decide half-way through some results that it got all it needs or it doesn't like what it sees and decide to close the connection.

Reply:
That's fair (I don't consider the single-thread case enough). I think what I want is a check for open results on `Pool.Close`.